src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -32,14 +52,52 @@ public class EditCommandParser implements Parser<EditCommand> {
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_SUBJECT, PREFIX_DAY, PREFIX_TIME, PREFIX_PHONE,
+                        PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_SCHOOL, PREFIX_NEXTOFKIN,
+                        PREFIX_LEVEL, PREFIX_QUALIFICATION, PREFIX_INSTITUTION);
 
         Index index;
 
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.getMessageUsage(listType)), pe);
+        }
+
+        EditDescriptor editDescriptor;
+
+        switch (listType) {
+        case STUDENT_LIST:
+            editDescriptor = extractFromMapForStudent(argMultimap);
+            break;
+
+        case TUTOR_LIST:
+            editDescriptor = extractFromMapForTutor(argMultimap);
+            break;
+
+        case TUITIONCLASS_LIST:
+            editDescriptor = extractFromMapForClass(argMultimap);
+            break;
+
+        default:
```

Comment:
Same as above, maybe can insert assert in default. Also, maybe shifting the TUITION_CLASS case to the default would be better as we would be deleting PERSON_LIST in the future.